### PR TITLE
snmp-ups: simplify Eaton ePDU group phase handling

### PR DESCRIPTION
**Pull request by aquette**
Signed-off-by: Arnaud Quette <ArnaudQuette@Eaton.com>

### CODE REVIEW

File: drivers/eaton-pdu-marlin-mib.c [hunk 1]
```diff
@@ -876,50 +849,15 @@ static snmp_info_t eaton_marlin_mib[] = {
...
+	/* Phase to which an outlet-group is connected:
+	 * We use the following OID, which gives the voltage measurement type
+	 * groupVoltageMeasType.0.1; Value (Integer): singlePhase  (1)
 	 */
 	/* FIXME: RFC on key name is needed when backporting to NUT upstream ; check type (number? string?) and flags (daisy?) */
 	{ "outlet.group.%i.phase", 0, SU_INFOSIZE,
-		".1.3.6.1.4.1.534.6.6.7.5.1.1.10.%i.%i",
+		".1.3.6.1.4.1.534.6.6.7.5.3.1.2.%i.%i",
 		NULL, SU_FLAG_UNIQUE | SU_FLAG_STATIC | SU_OUTLET_GROUP | SU_TYPE_DAISY_1,
 		&marlin_outlet_group_phase_info[1], NULL },
```
Comment (jimklimov) [inline drivers/eaton-pdu-marlin-mib.c]: Doesn't this map all `.%i.%i` templated values to the same 1-phase entry in the table?

File: drivers/eaton-pdu-marlin-mib.c [hunk 2]
```diff
@@ -876,50 +849,15 @@ static snmp_info_t eaton_marlin_mib[] = {
...
-	 * groupPhaseID:  // FIXME-Check on real device
+	/* Phase to which an outlet-group is connected:
+	 * We use the following OID, which gives the voltage measurement type
+	 * groupVoltageMeasType.0.1; Value (Integer): singlePhase  (1)
 	 */
 	/* FIXME: RFC on key name is needed when backporting to NUT upstream ; check type (number? string?) and flags (daisy?) */
 	{ "outlet.group.%i.phase", 0, SU_INFOSIZE,
-		".1.3.6.1.4.1.534.6.6.7.5.1.1.10.%i.%i",
+		".1.3.6.1.4.1.534.6.6.7.5.3.1.2.%i.%i",
 		NULL, SU_FLAG_UNIQUE | SU_FLAG_STATIC | SU_OUTLET_GROUP | SU_TYPE_DAISY_1,
```
Comment (jimklimov) [inline drivers/eaton-pdu-marlin-mib.c]: Also, no longer `SU_FLAG_UNIQUE`?

File: drivers/eaton-pdu-marlin-mib.c [hunk 3]
```diff
@@ -876,50 +849,15 @@ static snmp_info_t eaton_marlin_mib[] = {
...
-	 * (Caution: the value is numeric, and need to append "L" -- TODO)
-	 * groupPhaseID:  // FIXME-Check on real device
+	/* Phase to which an outlet-group is connected:
+	 * We use the following OID, which gives the voltage measurement type
+	 * groupVoltageMeasType.0.1; Value (Integer): singlePhase  (1)
 	 */
 	/* FIXME: RFC on key name is needed when backporting to NUT upstream ; check type (number? string?) and flags (daisy?) */
 	{ "outlet.group.%i.phase", 0, SU_INFOSIZE,
-		".1.3.6.1.4.1.534.6.6.7.5.1.1.10.%i.%i",
+		".1.3.6.1.4.1.534.6.6.7.5.3.1.2.%i.%i",
```
Comment (jimklimov) [inline drivers/eaton-pdu-marlin-mib.c]: Do we still have two or three OID entries of different preference to pick from, or just one (applicable for all firmware versions, including legacy) now?

File: drivers/eaton-pdu-marlin-mib.c [hunk 4]
```diff
@@ -175,27 +175,28 @@ static info_lkp_t marlin_input_type_info[] = {
...
+static info_lkp_t marlin_outlet_group_phase_info[] = {
+	{ 0, "unknown", NULL, NULL, NULL, NULL }, /* unknown     */
+	{ 1, "1", NULL, NULL, NULL, NULL },       /* singlePhase */
+	{ 2, "1-N", NULL, NULL, NULL, NULL },     /* phase1toN   */
+	{ 3, "2-N", NULL, NULL, NULL, NULL },     /* phase2toN   */
+	{ 4, "3-N", NULL, NULL, NULL, NULL },     /* phase3toN   */
+	{ 5, "1-2", NULL, NULL, NULL, NULL },     /* phase1to2   */
+	{ 6, "2-3", NULL, NULL, NULL, NULL },     /* phase2to3   */
+	{ 7, "3-1", NULL, NULL, NULL, NULL },     /* phase3to1   */
+	{ 0, NULL, NULL, NULL, NULL, NULL }
```
Comment (jimklimov) [inline drivers/eaton-pdu-marlin-mib.c]: Something in this table upset the DMF parser on Travis builds, will try to reproduce. Maybe some `is None` is missing in our Pythonware.

````
3523 I: Parsed /home/travis/build/42ity/nut/drivers/eaton-pdu-marlin-mib.c OK
3524 I: CALL Visitor():
3525 Traceback (most recent call last):
3526  File "/home/travis/build/42ity/nut/scripts/DMF/jsonify-mib.py", line 572, in <module>
3527    v.visit (node)
3528  File "/usr/local/lib/python2.7/dist-packages/pycparser/c_ast.py", line 120, in visit
3529    return visitor(node)
3530  File "/home/travis/build/42ity/nut/scripts/DMF/jsonify-mib.py", line 319, in visit_Decl
3531    self._visit_info_lkp_t (node)
3532  File "/home/travis/build/42ity/nut/scripts/DMF/jsonify-mib.py", line 205, in _visit_info_lkp_t
3533    debug("fun_l2s.name : %s" % (fun_l2s.name))
3534 AttributeError: 'Cast' object has no attribute 'name'
3535 ERROR: Could not parse '/home/travis/build/42ity/nut/drivers/eaton-pdu-marlin-mib.c' into 'eaton-pdu-marlin-mib.dmf'
3536       You can inspect a copy of the intermediate result in 'eaton-pdu-marlin-mib.json.tmp', 'eaton-pdu-marlin-mib.dmf.tmp' and 'eaton-pdu-marlin-mib_TEST.c'
3537 make[2]: *** [dmfsnmp/eaton-pdu-marlin-mib.dmf] Error 1
````

Comment (aquette) [reply drivers/eaton-pdu-marlin-mib.c]: strange, I did not had that one on my side

File: drivers/eaton-pdu-marlin-mib.c (hunk 3 above)
Comment (aquette) [reply drivers/eaton-pdu-marlin-mib.c]: just one now

File: drivers/eaton-pdu-marlin-mib.c (hunk 2 above)
Comment (aquette) [reply drivers/eaton-pdu-marlin-mib.c]: indeed, forgot to update. done now

File: drivers/eaton-pdu-marlin-mib.c (hunk 1 above)
Comment (aquette) [reply drivers/eaton-pdu-marlin-mib.c]: not sure to get your comment. potentially the erroneous [1] index. fixed now too

**Pull request merged**